src/unpack.c

```diff
@@ -1443,9 +1443,13 @@ void process_archive(const char *filename) {
   tar.ops = &tf;
 
   rc = tar_extractor(&tar);
-  if (rc)
+  if (rc){
     dpkg_error_print(&tar.err,
                      _("corrupted filesystem tarfile in package archive"));
+  }
+  if(tar.err.str) // todo: during normal operation no error strings should be set!
+    free(tar.err.str);
+
   if (fd_skip(p1[0], -1, &err) < 0)
```

Comment:
This didn't make sense. If tar.err is set but no error has been reported then something else is wrong. Otherwise there should be no need to free it. If you have found an instance where this happens, I'd appreciate a description of how/where that happens.